### PR TITLE
NEXUS-8305: APIs made public, added IT (Part 2)

### PR DESCRIPTION
**Pull request by cstamas**
Splitting original PR #1257 into multiple smaller PR "steps" for easier review. This is step 2, with changes:
* more moves of public API parts from `internal` package (MavenFacet, VersionPolicy).
* added "sanity" IT that exercises Maven repository

Issue
https://issues.sonatype.org/browse/NEXUS-8305

CI
http://bamboo.s/browse/NX3-OSSF310

### CODE REVIEW

**Comment by cstamas** [comment]
http://bamboo.s/browse/NX3-OSSF310-1 :+1:

File: testsuite/modern-testsuite/src/test/it-resources/org/sonatype/nexus/testsuite/maven/testplugin/src/main/java/org/sonatype/nexus/testsuite/MyMojo.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Goal which touches a timestamp file.
+ *
+ * @deprecated Don't use!
+ */
+@Mojo( name = "touch", defaultPhase = LifecyclePhase.PROCESS_SOURCES )
```
Comment (mrprescott) [inline testsuite/modern-testsuite/src/test/it-resources/org/sonatype/nexus/testsuite/maven/testplugin/src/main/java/org/sonatype/nexus/testsuite/MyMojo.java]: Do we leave original formatting on imported code?

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/MavenITSupport.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      // set POM version
+      final String pomXml = Files.toString(new File(projectDir, "pom.xml"), Charsets.UTF_8).replace(
+          "${project.version}", version);
+      Files.write(pomXml, mavenPom, Charsets.UTF_8);
+    }
+
+    Verifier verifier = new Verifier(mavenBaseDir.getAbsolutePath());
+    verifier.addCliOption("-s " + mavenSettings.getAbsolutePath());
+    verifier.addCliOption(
+        // TODO: verifier replaces // -> /
```
Comment (mrprescott) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/MavenITSupport.java]: Why a TODO?

**Comment by mrprescott** [comment]
+1, minor comments

File: testsuite/modern-testsuite/src/test/it-resources/org/sonatype/nexus/testsuite/maven/testplugin/src/main/java/org/sonatype/nexus/testsuite/MyMojo.java (hunk 1 above)
Comment (cstamas) [inline testsuite/modern-testsuite/src/test/it-resources/org/sonatype/nexus/testsuite/maven/testplugin/src/main/java/org/sonatype/nexus/testsuite/MyMojo.java]: This s maven quickstart archetype as-is, and basically we don't care about it's contents, we want maven to deploy it... I think it's ok to leave it as is, maybe a README in root to describe from where these sources come (maven-quickstart-archetype:1.1 I think)

File: testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/MavenITSupport.java (hunk 2 above)
Comment (cstamas) [inline testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/MavenITSupport.java]: Ack

**Comment by cstamas** [comment]
Merged as 77cd0b381a9cda6e88056931554c1e2c9d3cf444

**Pull request closed without merging**